.github/workflows/shark_model_suite.yml

```diff
@@ -0,0 +1,91 @@
+          # See https://github.com/actions/setup-python/issues/433
+          echo ${{ matrix.python-version }} >> $GITHUB_WORKSPACE/.python-version
+      
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: '${{ matrix.python-version }}'
+          #cache: 'pip'
+          #cache-dependency-path: |
+          #  **/requirements-importer.txt
+          #  **/requirements.txt
+      - name: Download SHARK
+        id: download_shark
+        uses: actions/checkout@v2
+        with:
+          repository: nod-ai/SHARK
+          path: SHARK
+          ref: 20220708.119
+      - name: Download packages
+        id: download_packages
+        uses: dawidd6/action-download-artifact@b2abf1705491048a2d7074f7d90513044fd25d39  # v2
+        with:
+          github_token: ${{secrets.WRITE_ACCESS_TOKEN}}
+          workflow: build_package.yml
+          run_id: ${{ github.event.inputs.build_run_id }}
+      - name: Extract and display downloaded files
+        run: |
+          tar -xf artifact/iree-dist-${{ github.event.inputs.package_version }}-linux-x86_64.tar.xz
+          pwd
+          ls -R
+      - name: Validate CPU Models
+        if: matrix.suite == 'cpu'
+        run: |
+          cd $GITHUB_WORKSPACE
+          cd SHARK
+          PYTHON=python${{ matrix.python-version }} USE_IREE=1 IMPORTER=1 ./setup_venv.sh
+          source shark.venv/bin/activate
+          python -m pip install --force-reinstall -f file://$PWD/artifact/ iree-compiler iree-runtime iree-tools-tflite iree-tools-tf iree-tools-xla
+          pytest -n 2 -k 'not benchmark'  --ignore=shark/tests/test_shark_importer.py --ignore=tank/tf/  --ignore=benchmarks/tests/test_hf_benchmark.py --ignore=benchmarks/tests/test_benchmark.py 
+      - name: Validate GPU Models
+        if: matrix.suite == 'gpu'
+        run: |
+          cd $GITHUB_WORKSPACE
+          cd SHARK
+          PYTHON=python${{ matrix.python-version }} USE_IREE=1 IMPORTER=1 ./setup_venv.sh
+          source shark.venv/bin/activate
+          python -m pip install --force-reinstall -f file://$PWD/artifact/ iree-compiler iree-runtime iree-tools-tflite iree-tools-tf iree-tools-xla
+          pytest -k 'gpu' --ignore=shark/tests/test_shark_importer.py --ignore=benchmarks/tests/test_hf_benchmark.py --ignore=benchmarks/tests/test_benchmark.py 
+      - name: Validate Vulkan Models
+        if: matrix.suite == 'vulkan'
+        run: |
+          cd $GITHUB_WORKSPACE
+          cd SHARK
+          PYTHON=python${{ matrix.python-version }} USE_IREE=1 IMPORTER=1 ./setup_venv.sh
+          source shark.venv/bin/activate
+          python -m pip install --force-reinstall -f file://$PWD/artifact/ iree-compiler iree-runtime iree-tools-tflite iree-tools-tf iree-tools-xla
+          pytest -k 'vulkan' --ignore=shark/tests/test_shark_importer.py --ignore=benchmarks/tests/test_hf_benchmark.py --ignore=benchmarks/tests/test_benchmark.py
```

Comment:
These steps are almost entirely the same. Can we inject the differences as variables rather than repeating the steps 3 times? At least, it seems like we should be able to run the first 5 commands as a separate step ahead of time